x-pack/plugins/session_view/public/components/detail_panel_process_tab/helpers.ts

```diff
@@ -46,3 +68,86 @@ export const formatProcessArgs = (args: string[]): string =>
  */
 export const getIsInterativeString = (tty: Teletype | undefined): string =>
   !!tty ? 'True' : 'False';
+
+const getDetailPanelProcessLeader = (
+  leader: ProcessFields | undefined
+): DetailPanelProcessLeader => ({
+  ...leader,
+  id: leader?.entity_id ?? DEFAULT_PROCESS_DATA.id,
+  name: leader?.name ?? DEFAULT_PROCESS_DATA.name,
+  start: leader?.start ?? DEFAULT_PROCESS_DATA.start,
+  end: leader?.end ?? DEFAULT_PROCESS_DATA.end,
+  exitCode: leader?.exit_code?.toString() ?? DEFAULT_PROCESS_DATA.exitCode,
+  interactive: getIsInterativeString(leader?.tty),
+  userName: leader?.user?.name ?? DEFAULT_PROCESS_DATA.userName,
+  groupName: leader?.group?.name ?? DEFAULT_PROCESS_DATA.groupName,
+  workingDirectory: leader?.working_directory ?? DEFAULT_PROCESS_DATA.workingDirectory,
+  args: formatProcessArgs(leader?.args) ?? DEFAULT_PROCESS_DATA.args,
+  pid: leader?.pid?.toString() ?? DEFAULT_PROCESS_DATA.pid,
+  // TODO: get the event action of leader
+  executable: leader?.executable ? [[leader?.executable]] : DEFAULT_PROCESS_DATA.executable,
+  entryMetaType: leader?.entry_meta?.type ?? DEFAULT_PROCESS_DATA.entryMetaType,
+  entryMetaSourceIp: leader?.entry_meta?.source?.ip ?? DEFAULT_PROCESS_DATA.entryMetaSourceIp,
+});
+
+export const getDetailPanelProcess = (process: Process | null): DetailPanelProcess => {
+  const processData = {
+    id: DEFAULT_PROCESS_DATA.id,
+    start: DEFAULT_PROCESS_DATA.start,
+    end: DEFAULT_PROCESS_DATA.end,
+    exitCode: DEFAULT_PROCESS_DATA.exitCode,
+    interactive: DEFAULT_PROCESS_DATA.interactive,
+    userName: DEFAULT_PROCESS_DATA.userName,
+    groupName: DEFAULT_PROCESS_DATA.groupName,
+    args: DEFAULT_PROCESS_DATA.args,
+    pid: DEFAULT_PROCESS_DATA.pid,
+    executable: DEFAULT_PROCESS_DATA.executable,
+    workingDirectory: DEFAULT_PROCESS_DATA.workingDirectory,
+    entryLeader: DEFAULT_PROCESS_DATA,
+    sessionLeader: DEFAULT_PROCESS_DATA,
+    groupLeader: DEFAULT_PROCESS_DATA,
+    parent: DEFAULT_PROCESS_DATA,
+  } as DetailPanelProcess;
+  if (!process) {
+    return processData;
+  }
+
+  const details = process.getDetails();
+
+  processData.id = `${dataOrDash(process.id)}`;
+  processData.start = `${dataOrDash(details.process?.start)}`;
+  processData.end = `${dataOrDash(process.getEndTime())}`;
+  processData.exitCode = `${dataOrDash(details.process?.exit_code)}`;
+  processData.interactive = getIsInterativeString(details.process?.tty);
```

Comment:
does the absence of `process.interactive` will make it show Interactive as 'False'? I think in case of absence it should show dash to avoid misunderstood of the information

Reply:
I remember Karl mentioned before that we should be using `process.tty` instead of `process.interactive`, I think the reason is [here](https://github.com/elastic/ecs/blob/main/rfcs/text/0030/process.yml#L64):
> Note: A non-interactive process can belong to an interactive session and is simply one that does not have open file descriptors reading the controlling TTY on FD 0 (stdin) or writing to the controlling TTY on FD 2 (stderr). A backgrounded process is still considered interactive if stdin and stderr are connected to the controlling TTY.

If I remember correctly, the process is then considered interactive if `process.tty` exists, and non-interactive if it doesn't exist.

cc @mitodrummer

Reply:
should we change it to be `process.tty` in the Details Panel then?

Reply:
also, if by any change `process.tty` did not get collected by Endpoint, we will still want to show as `False`? I'm afraid that could confuse the user if the user tries to query `proces.tty: false` on EQL

Reply:
process.tty is the controlling tty for the session as a whole. Processes that belong to an "interactive session (has tty)" can be run in the background, or not be connected to the process group leaders tty, hence the separate 'interactive' boolean on the process fieldset.

Currently we aren't showing the tty info in the details panel, but we probably should. Might be worth consulting with design on how best to represent the major/minor device chars fields that exist as part of it. cc @codearos

Reply:
I am unsure if process.tty is set, when process.interactive is false. I would assume process.tty is always set if the session has a ctty. Something to verify I guess.

Reply:
cc @m-sample for accuracy in my above statements :)

Reply:
> also, if by any change `process.tty` did not get collected by Endpoint, we will still want to show as `False`? I'm afraid that could confuse the user if the user tries to query `proces.tty: false` on EQL

process.tty is not a boolean. it's an object with { major, minor } device numbers. A query to find all processes that belong to an interactive session would look like    process.tty: *

Reply:
The absense of process.tty means the process belongs to a non interactive session.

Reply:
I see, that all makes sense, thanks for clarifying everything @mitodrummer

Reply:
Karl - you called it; interactive *sessions* have an associated TTY.  It should always be there for process events interactive sessions (*iff* widening is enabled).   Tty  fieldset will get more fields in the future (terminal type, tty settings like utf-8 etc).

Interactive processes may only occur in interactive sessions, but not all processes in an interactive session are marked as interactive processe - only those whose stdin and stderr are bound to TTY.

Careful with assuming implicit values for fields absent from an event.   For example if interactive_process is absent in the event but widening is off, its value is *unknown* (could be either true or false, we didn't record it).   If widening is on, it should likely be interpreted as false (aligns with Go's concept of "zero values")